File: Java/day10.java
import java.util.*;

public class day10
{
    public static void main(String[] args) {
        // Write your code here
        Scanner sc = new Scanner(System.in);
        int T = sc.nextInt();
        for (int i = 0; i < T; i++) {
            int arg = sc.nextInt(), divisor = 2;
            int quotient = arg;
            Stack<Integer> bin = new Stack<Integer>();
            while (quotient > 0) {
                int mod = quotient % divisor;
                bin.push(mod);
                quotient /= divisor;
            }
            for (int j = bin.size()-1; j >= 0; j--)
                System.out.print(bin.get(j));
            System.out.println();
        }
    }
}

File: Scala/day10.scala
import scala.io.StdIn // added to avoid a deprecation warning for readLine()
import scala.collection.mutable.Stack

object Solution {
  def main(args: Array[String]) {
    // Write your code here
    val t = StdIn.readLine()
    for (_ <- 1 to t.toInt) {
      val dec = StdIn.readLine()
      val (arg, divisor) = (dec.toInt, 2)
      var quotient = arg
      var bin = Stack[Int]()
      while (quotient > 0) {
        val mod = quotient % divisor
        bin.push(mod)
        quotient /= divisor
      }
      bin.foreach(print)
      println()
    }
  }
}
